src/map/battle.cpp

```diff
@@ -3353,25 +3356,67 @@ static bool attack_ignores_def(struct Damage* wd, struct block_list *src, struct
 	return nk[NK_IGNOREDEFENSE] != 0;
 }
 
-/*================================================
- * Should skill attack consider VVS and masteries?
- *------------------------------------------------
- * Credits:
- *	Original coder Skotlex
- *	Initial refactoring by Baalberith
- *	Refined and optimized by helvetica
+/**
+ * This function lists which skills are unaffected by refine bonus, masteries, Star Crumbs and Spirit Spheres
+ * This function is also used to determine if atkpercent applies
+ * @param skill_id: Skill being used
+ * @param type 1 - Checking refine bonus; 2 - Checking Star Crumb bonus
+ * @return true = bonus applies; false = bonus does not apply
  */
-static bool battle_skill_stacks_masteries_vvs(uint16 skill_id)
+static bool battle_skill_stacks_masteries_vvs(uint16 skill_id, int type)
 {
-	if (
-#ifndef RENEWAL
-		skill_id == PA_SHIELDCHAIN || skill_id == CR_SHIELDBOOMERANG ||
-		skill_id == PA_SACRIFICE || skill_id == AM_ACIDTERROR ||
-#endif
-		skill_id == MO_INVESTIGATE || skill_id == MO_EXTREMITYFIST ||
-		skill_id == RK_DRAGONBREATH || skill_id == RK_DRAGONBREATH_WATER || skill_id == NC_SELFDESTRUCTION ||
-		skill_id == LG_SHIELDPRESS || skill_id == LG_EARTHDRIVE)
+	switch (skill_id) {
+	case PA_SHIELDCHAIN:
+	case CR_SHIELDBOOMERANG:
+	case AM_ACIDTERROR:
+	case MO_INVESTIGATE:
+	case MO_EXTREMITYFIST:
+	case PA_SACRIFICE:
+	case NPC_DRAGONBREATH:
+	case RK_DRAGONBREATH:
+	case RK_DRAGONBREATH_WATER:
+	case NC_SELFDESTRUCTION:
+	case LG_SHIELDPRESS:
+	case LG_EARTHDRIVE:
+		return false;
+	case CR_GRANDCROSS:
+	case NPC_GRANDDARKNESS:
+		// Grand Cross is influenced by refine bonus but not by atkpercent / masteries / Star Crumbs / Spirit Spheres
+		if (type != 1)
+			return false;
+		break;
+	case LK_SPIRALPIERCE:
+		// Spiral Pierce is influenced only by refine bonus and Star Crumbs
+		if (type != 1 && type != 2)
+			return false;
+		break;
+	}
+
+	return true;
+}
+
+/**
+ * This function lists which skills are unaffected by EDP and the elemental bonus from Magnum Break / EDP
+ * Unit skills (e.g. Bomb and Freezing Trap) are never affected.
+ * @param skill_id: Skill being used
+ * @return true = bonus applies; false = bonus does not apply
+ */
+static bool battle_skill_stacks_edp_element(uint16 skill_id)
```

Comment:
May be just me, but this whole function sounds like a flag that should be put into skill_db.yml

Reply:
Same as above, I won't mess with YML myself, but it's something that can be done later, yes.

Reply:
Feature request added: https://github.com/rathena/rathena/issues/8226